source/s3_request_messages.c

```diff
@@ -393,6 +393,17 @@ struct aws_http_message *aws_s3_message_util_copy_http_message(
         if (aws_http_message_add_header(message, header)) {
             goto error_clean_up;
         }
+
+        /* For SSE upload, the sse related headers should only be shown in the create_multipart_upload. which means it
+         * should only be copied once.
+         * TODO: should be all the headers begin with x-amz-server-side-encryption-* ? */
+        if (aws_byte_cursor_eq_c_str_ignore_case(&header.name, "x-amz-server-side-encryption")) {
```

Comment:
This functionality should be opt-in and not touch the base_message.  (base_message should already be const for this function.  That's my mistake!)  What I think we want to do differently is just not copy these headers to the new message if the caller passes in a flag to this function.  And then we can pass in that flag where ever we need to in any of the existing places where we use this function.